Sources/TokamakCore/Views/Gestures/_GestureView.swift

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Szymon on 16/7/2023.
+//
+
+import Foundation
+
+public struct _GestureView<Content: View, G: Gesture>: _PrimitiveView {
+    @Environment(\.isEnabled) var isEnabled
+    @Environment(\.gestureListener) var gestureListener
+    @State public var gestureId: String = UUID().uuidString
+    @State public var gesture: G
+    @State var eventId: String? = nil
+    
+    let mask: GestureMask
+    let priority: _GesturePriority
+    public let content: Content
+    
+    var minimumDuration: Double? {
+        guard let longPressGesture = gesture as? LongPressGesture else {
+            return nil
+        }
+        return longPressGesture.minimumDuration
+    }
+    
+    public init(
+        gesture: G,
+        mask: GestureMask,
+        priority: _GesturePriority = .standard,
+        content: Content
+    ) {
+        self._gesture = State(wrappedValue: gesture)
+        self.mask = mask
+        self.priority = priority
+        self.content = content
+    }
+    
+    public func onPhaseChange(_ phase: _GesturePhase, eventId id: String? = nil) {
+        guard isEnabled, let currentEventId = eventId ?? id else { return }
+        
+        let value = GestureValue(gestureId: gestureId, mask: mask, priority: priority)
+
+        switch phase {
+        case .began:
+            startDelay()
+            eventId = id
+            gestureListener.registerStart(value, for: currentEventId)
+        case .cancelled, .ended:
+            eventId = nil
+        default:
+            break
+        }
+        
+        guard gestureListener.canProcessGesture(value, for: currentEventId) else {
+            // Event being processed by another gestures
+            return
+        }
+        
+        if gesture._onPhaseChange(phase) {
+            gestureListener.recognizeGesture(value, for: currentEventId)
+        }
+    }
+    
+    private func startDelay() {
+        guard let minimumDuration else { return }
+        Task {
+            do {
+                try await Task.sleep(for: .seconds(minimumDuration))
+                if let eventId {
+                    await MainActor.run {
+                        onPhaseChange(.changed(location: nil), eventId: eventId)
+                    }
+                }
+            } catch {
+                //TODO: What do we do with this error?
+                print(error)
```

Comment:
I think it would only fail if the task is cancelled before the `Task.sleep` finished. So it might be ok to ignore the error instead of logging it?
```suggestion
```

Reply:
fixed